app/services/base_index_job.rb

```diff
@@ -3,24 +3,46 @@ class BaseIndexJob
               :book_version_id,
               :type
 
-  def initialize(book_version_id:, indexing_strategy_name:, when_completed_proc: nil)
-    @when_completed_proc = when_completed_proc
+  def self.build_object(params:, cleanup_after_call: nil)
+    new(book_version_id:        params[:book_version_id],
+        indexing_strategy_name: params[:indexing_strategy_name],
+        cleanup_after_call:     cleanup_after_call)
+  end
+
+  def initialize(book_version_id: nil,
+                 indexing_strategy_name: nil,
+                 cleanup_after_call: nil)
+    @cleanup_after_call = cleanup_after_call
 
     @type = self.class.to_s
     @book_version_id = book_version_id
     @indexing_strategy_name = indexing_strategy_name
+  end
+
+  def cleanup_after_call
+    @cleanup_after_call.try(:call)
+  end
+
+  def cleanup_when_done
+  end
 
-    @book_uuid, @book_version = @book_version_id.split('@')
+  def remove_associated_book_index
+    book_index = find_associated_book_index
+    book_index.destroy!
   end
 
-  def when_completed
-    @when_completed_proc.try(:call)
+  def metadata
+    internal_data = JSON.parse(to_json)
+    book_index = find_associated_book_index
```

Comment:
per below, make the var `book_index_state`.

You could also consider just making the method be `book_index_state`

Reply:
renamed the method inspect